src/settings.rs

```diff
@@ -27,22 +34,176 @@ impl<'src> Settings<'src> {
   }
 
   pub(crate) fn shell_binary<'a>(&'a self, config: &'a Config) -> &'a str {
-    if let (Some(shell), false) = (&self.shell, config.shell_present) {
+    // Order of precedence:
+    // 1) CLI argument (config)
+    // 2) Settings in justfile (settings)
+    // 3) Default/PowerShell
```

Comment:
I think this is mostly evident from reading the code, so I'd remove this comment.

```suggestion
```